stock_request/models/stock_request.py

```diff
@@ -97,14 +108,29 @@ def _get_default_requested_by(self):
         states={'draft': [('readonly', False)]}, readonly=True
     )
     route_id = fields.Many2one(
-        states={'draft': [('readonly', False)]}, readonly=True
+        states={'draft': [('readonly', False)],
+                'submitted': [('readonly', False)]}, readonly=True
     )
+    direction = fields.Selection([('outbound', 'Outbound'),
+                                  ('inbound', 'Inbound')],
+                                 string='Direction',
+                                 states={'draft': [('readonly', False)]},
+                                 readonly=True)
 
     _sql_constraints = [
         ('name_uniq', 'unique(name, company_id)',
          'Stock Request name must be unique'),
     ]
 
+    @api.onchange('direction')
```

Comment:
As I said, stock requests should be used for internal movements. If you think it is interesting for outside movements, maybe a stock_request_sale module should be necessary

Reply:
I don't understand why you want to limit stock requests to internal moves when it can apply to a broader scope...

Reply:
Because it is not a sale. It will not generate invoices or sale orders.

Reply:
I am not sure to understand why it matters for the stock request module. Even an internal move can generate others documents, for example an accounting entry. I had a customer who tracked 2 internal locations in 2 different financial accounts.

Reply:
I think that the field 'direction' is very opinionated. A normal user would not understand why is this needed. IMHO this should be a customer-specific customization.